src/toil/jobStores/fileJobStore.py

```diff
@@ -409,16 +409,27 @@ def writeFile(self, localFilePath, jobStoreID=None, cleanup=False):
         return relPath
 
     @contextmanager
-    def writeFileStream(self, jobStoreID=None, cleanup=False, basename=None):
+    def writeFileStream(self, jobStoreID=None, cleanup=False, basename=None, mode='b',
+                        encoding=None, errors=None):
         if not basename:
             basename = 'stream'
         absPath = self._getUniqueFilePath(basename, jobStoreID, cleanup)
         relPath = self._getFileIdFromPath(absPath)
-        with open(absPath, 'wb') as f:
-            # Don't yield while holding an open file descriptor to the temp
-            # file. That can result in temp files still being open when we try
-            # to clean ourselves up, somehow, for certain workloads.
-            yield f, relPath
+
+        if mode == 'b':
+            with open(absPath, 'wb', errors=errors) as f:
+                # Don't yield while holding an open file descriptor to the temp
+                # file. That can result in temp files still being open when we try
+                # to clean ourselves up, somehow, for certain workloads.
+                yield f, relPath
+
+        elif mode == 't':
+            with open(absPath, 'w', encoding=encoding, errors=errors) as f:
+                # Don't yield while holding an open file descriptor to the temp
+                # file. That can result in temp files still being open when we try
+                # to clean ourselves up, somehow, for certain workloads.
+                yield f, relPath
+
```

Comment:
What if someone passed `mode='wombats'`? It looks like that won't produce an informative error.